### ISSUE TITLE
Upload to Mac App Store for Safari support

### ISSUE THREAD

**Issue by scratchusernamemrtbts** [ScratchAddons/ScratchAddons#340]
**Is your feature request related to a problem? Please describe.**

some people that only use safari can't use this addon
**Describe the solution you'd like**

upload this to mac appstore to be a safari extension
**Describe alternatives you've considered**

**Additional context**

**Comment by JeffaloBob** [comment]
does safari support the same kind of extensions as chrome and firefox? @WorldLanguages

**Comment by scratchusernamemrtbts** [comment]
Yes

**Comment by scratchusernamemrtbts** [comment]
But you’ll need to make a home page for it as an app

**Comment by Thingyman-DEV** [comment]
I don't think they can. To make it actually on the mac store, you need an apple developer account which is _99 dollars a year_

**Comment by GrahamSH-LLK** [comment]
Yeah. I don't think we have that kind of money. If you really want it, just download chrome.

**Comment by WorldLanguages** [comment]
Two things missing:
- $99 a year
- Support for some APIs on Safari 14, like `chrome.cookies.addListener("change" ...` (source: [MDN](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies/onChanged))

**Comment by scratchusernamemrtbts** [comment]
I already try to convert it to safari
but there is still a bug
the popup page don't show the info it shows only the menu bar
but it works in the editor or when I click the popout
<img width="2560" alt="ภาพถ่ายหน้าจอ 2563-11-01 เวลา 07 53 43" src="https://user-images.githubusercontent.com/66320961/97792926-663c8580-1c17-11eb-9b5d-ebbdbf2ef335.png">
<img width="2560" alt="ภาพถ่ายหน้าจอ 2563-11-01 เวลา 07 54 18" src="https://user-images.githubusercontent.com/66320961/97792930-75233800-1c17-11eb-8d01-d3a27c3a3c11.png">


<img width="2560" alt="ภาพถ่ายหน้าจอ 2563-11-01 เวลา 07 44 57" src="https://user-images.githubusercontent.com/66320961/97792824-2628d300-1c16-11eb-87d6-7562a9184022.png">

<img width="2560" alt="ภาพถ่ายหน้าจอ 2563-11-01 เวลา 07 42 06" src="https://user-images.githubusercontent.com/66320961/97792797-c29ea580-1c15-11eb-85a8-3e1072281132.png">

**Comment by scratchusernamemrtbts** [comment]
if someone have a Mac can someone try it on there Mac
[Scratch Addons app](https://github.com/ScratchAddons/ScratchAddons/files/5470055/Scratch.Addons.zip)
## how to enable the add-on
1. turn on developer mode in safari advance setting
2. then click the develop menu in the top left corner
3. click allow unsigned extension
4. enter your Mac password
5. then press enter
6. launch the scratch add-on app
7. then click Quit and Open Safari Extensions Preferences…
8. check the Scratch Addons box

when you quit safari the allow unsigned extension setting will be reset and you need to do  step 2-5 again

**Comment by YOYITsM3M8** [comment]
wait so we dont need to pay $99 a year

**Comment by GrahamSH-LLK** [comment]
Well, that's debatable. You have to set it again every single time you start safari.

**Comment by YOYITsM3M8** [comment]
Oh

**Comment by scratchusernamemrtbts** [comment]
Can someone reopen this issue

**Comment by GrahamSH-LLK** [comment]
> Can someone reopen this issue

No. We're not going to either A. pay 99 a year, or B. Make it so that you have to re-set it every time safari starts.

**Comment by YOYITsM3M8** [comment]
10/10 method keep safari open

**Comment by easrng** [comment]
Someone make an applescript that installs SA using accesibility actions and make it run every time safari opens.

**Comment by scratchusernamemrtbts** [comment]
I'm searching for a way to make safari to allow unsigned extension and doesn't reset when quit

**Comment by scratchusernamemrtbts** [comment]
> 10/10 method keep safari open

@YOYITsM3M8 
if we shut down the computer it will quit safari

**Comment by YOYITsM3M8** [comment]
> Someone make an applescript that installs SA using accesibility actions and make it run every time safari opens.

but the addon settings wont save

**Comment by Thingyman-DEV** [comment]
Has anything changed as of big sur, or is it still not possible?

**Comment by GrahamSH-LLK** [comment]
> Has anything changed as of big sur, or is it still not possible?

Not as far as I know. You can get it to load using 
- turn on developer mode in safari advance setting
- then click the develop menu in the top left corner
- click allow unsigned extension
- enter your Mac password
- then press enter
- launch the scratch add-on app
- then click Quit and Open Safari Extensions Preferences…
- check the Scratch Addons box
- when you quit safari the allow unsigned extension setting will be reset and you need to do step 2-8 again

**Comment by WorldLanguages** [comment]
lol if Scratch Addons had a patreon, $9/month of contributions would be enough for the domain and App Store fee, just saying :P

**Comment by Hans5958** [comment]
do it, set up a gh sponsor also

**Comment by TheColaber** [comment]
time to make money 🤑🤑🤑🤑🤑

**Comment by scratchusernamemrtbts** [comment]
> Someone make an applescript that installs SA using accesibility actions and make it run every time safari opens.

done it. it doesn't run automatically but it works
[tmp.scpt.zip](https://github.com/ScratchAddons/ScratchAddons/files/5819026/tmp.scpt.zip)
here is a zip file

**Comment by scratchusernamemrtbts** [comment]
someone chack it please?

**Comment by easrng** [comment]
> here is a zip file
Can you send a plain `.applescript` file please?

**Comment by scratchusernamemrtbts** [comment]
> > here is a zip file
> 
> Can you send a plain `.applescript` file please?

no github doesn't support uploading it and the file extension is `.scpt` not `.applescript`

**Comment by easrng** [comment]
> > > here is a zip file
> > 
> > 
> > Can you send a plain `.applescript` file please?
> 
> no github doesn't support uploading it and the file extension is `.scpt` not `.applescript`

Do File > Save As > Plain Text, and it will save as a plain text `.applescript` file, instead of a compiled `.scpt`

**Comment by scratchusernamemrtbts** [comment]
> > > > here is a zip file
> > > 
> > > 
> > > Can you send a plain `.applescript` file please?
> > 
> > 
> > no github doesn't support uploading it and the file extension is `.scpt` not `.applescript`
> 
> Do File > Save As > Plain Text, and it will save as a plain text `.applescript` file, instead of a compiled `.scpt`

thank you

**Comment by scratchusernamemrtbts** [comment]
it still dosen't support that file
can i change it to `.txt`

**Comment by scratchusernamemrtbts** [comment]
[tmp.txt](https://github.com/ScratchAddons/ScratchAddons/files/5825391/tmp.txt)
here is the txt file

**Comment by scratchusernamemrtbts** [comment]
but the file is still make for use only on my thai mac
if i change it to english it wouldn't work on my mac because i'm using thai language

**Comment by scratchusernamemrtbts** [comment]
english version of the file
[tmp.txt](https://github.com/ScratchAddons/ScratchAddons/files/5825399/tmp.txt)

**Comment by ilikecereal1** [comment]
has anything changed with ventura

**Comment by GrahamSH-LLK** [comment]
> has anything changed with ventura

No 🙄

**Comment by Secret-chest** [comment]
Maybe something _will_ change, now that we have the EU Digital Markets act. It forces apple to:
* allow installation of software from any source
* allow third-party app/extension stores
* treat other platforms equally to their ones

**Comment by WorldLanguages** [comment]
> allow third-party app/**extension stores**

@Secret-chest Do you have a reliable source for the "extension stores" part?

**Comment by Secret-chest** [comment]
> > allow third-party app/**extension stores**
> 
> @Secret-chest Do you have a reliable source for the "extension stores" part?

I don't, but it seems plausible since they're also markets.

**Comment by zaid1442011** [comment]
> if someone have a Mac can someone try it on there Mac [Scratch Addons app](https://github.com/ScratchAddons/ScratchAddons/files/5470055/Scratch.Addons.zip)
> ## how to enable the add-on
> 
>     1. turn on developer mode in safari advance setting
> 
>     2. then click the develop menu in the top left corner
> 
>     3. click allow unsigned extension
> 
>     4. enter your Mac password
> 
>     5. then press enter
> 
>     6. launch the scratch add-on app
> 
>     7. then click Quit and Open Safari Extensions Preferences…
> 
>     8. check the Scratch Addons box
> 
> 
> when you quit safari the allow unsigned extension setting will be reset and you need to do step 2-8 again

It works, but the extention is outdated. Can someone make another one. (I maybe make one when I have time)

**Comment by Hans5958** [comment]
Reopening due to a possibility.

Discussion also happening on #3452.

**Comment by sfederici** [comment]
> https://github.com/ScratchAddons/ScratchAddons/pull/7006

I would love testing this, but I cannot find a way to checkout the code :-}

> if someone have a Mac can someone try it on there Mac [Scratch Addons app](https://github.com/ScratchAddons/ScratchAddons/files/5470055/Scratch.Addons.zip)

It installs fine and I can see the settings page. I tried enabling the "mouse position" and "stage on left side" addons but nothing shows up in the Scratch online editor.
